docs/articles/documentation/extending-testcafe/custom-browser-provider-plugin/index.md

```diff
@@ -0,0 +1,200 @@
+The generator will scaffold out the plugin, so that you only need to write a few lines of code.
+
+First, install [Yeoman](http://yeoman.io) and `generator-testcafe-browser-provider` using [npm](https://www.npmjs.com/).
+
+```bash
+npm install -g yo
+npm install -g generator-testcafe-browser-provider
+```
+
+Create a new directory where the generator should place your scaffolded project files and go into it.
+
+```bash
+mkdir my-provider
+cd my-provider
+```
+
+> It is recommended to name the directory as you would like to name the browser provider project. When you run the generator, it will automatically suggest the browser provider name that matches the browser provider directory name.
+>
+> The generator will also automatically create the provider package name that consists of two parts - the `testcafe-browser-provider-` prefix and the name of the provider itself, for example, `testcafe-browser-provider-my-provider`.
+>
+> **Important:** If you name the provider package manually, it is required that its name starts with the `testcafe-browser-provider-` prefix. Otherwise, TestCafe will be unable to recognize the plugin.
+
+Then run the browser provider generator to create a new project:
+
+```bash
+yo testcafe-browser-provider
+```
+
+The generator will ask you for information about the browser provider.
+Then Yeoman will automatically scaffold out your browser provider, install the required dependencies, and pull in several useful Gulp tasks for your workflow.
+
+## Implementing the Browser Provider
+
+Once the browser provider has been scaffolded out, go to the provider directory and open the `src/index.js` file.
+
+There are several required and optional provider methods you can implement.
+
+```js
+// Required methods
+async openBrowser (/* id, pageUrl, browserName */) {
+    throw new Error('Not implemented!');
+},
+
+async closeBrowser (/* id */) {
+    throw new Error('Not implemented!');
+},
+
+// Optional initialization and cleanup methods
+async init () {
+    return;
+},
+
+async dispose () {
+    return;
+},
+
+// Optional methods for multi-browser support
+async getBrowserList () {
+    throw new Error('Not implemented!');
+},
+
+async isValidBrowserName (/* browserName */) {
+    return true;
+},
+
+// Optional extra methods
+async resizeWindow (/* id, width, height, currentWidth, currentHeight */) {
+    this.reportWarning('The window resize functionality is not supported by the "my-provider" browser provider.');
+},
+
+async takeScreenshot (/* id, screenshotPath, pageWidth, pageHeight */) {
+    this.reportWarning('The screenshot functionality is not supported by the "my-provider" browser provider.');
+}
+```
+
+To implement these methods, you can use the [testcafe-browser-tools](https://github.com/DevExpress/testcafe-browser-tools) library.
+
+For details, see [Browser Provider Methods](browser-provider-methods.md).
```

Comment:
both these sentences are unclear. 

what does `testcafe-browser-tools` give me? I want to know why you want me to click the link.

for details about what? about `testcafe-browser-tools`? 'cause we were talking about `testcafe-browser-tools` a second ago!